src/audio/pipeline.c

```diff
@@ -562,13 +562,16 @@ int pipeline_reset(struct pipeline *p, struct comp_dev *host)
 static int pipeline_comp_copy(struct comp_dev *current, void *data, int dir)
 {
 	struct pipeline_data *ppl_data = data;
+	int is_single_ppl = comp_is_single_pipeline(current, ppl_data->start);
+	int is_same_sched =
+		pipeline_is_same_sched_comp(current->pipeline,
+					    ppl_data->start->pipeline);
 	int err = 0;
 
 	tracev_pipe("pipeline_comp_copy(), current->comp.id = %u, dir = %u",
 		    current->comp.id, dir);
 
-	if (!comp_is_single_pipeline(current, ppl_data->start) ||
-	    !comp_is_active(current)) {
+	if ((!is_single_ppl && !is_same_sched) || !comp_is_active(current)) {
```

Comment:
Far nicer to split this logic into single tests per condition, that can report a different trace to help with debug/developers.

Reply:
@tlauda as I understand this, if there are 2 ways to walk the pipeline, both the comp connected to the current comp will be copied. Is that the intention? For example, a mux can have 2 components connected at the output, this change will enabled copy to both the output components but how will you know which one should be picked?

Reply:
@ranj063 This change only enables the copy across mutliple pipelines, which share the same scheduling component. The best example would be mixer topology, where we have 2 host pipelines and one mixer pipeline with dai. Those 2 host pipelines don't have the scheduling components themselves, so we need to make sure the data will be copied properly.

Reply:
@tlauda thanks. that makes sense.